src/api.ts

```diff
@@ -100,33 +100,76 @@ export async function deleteImage(hash: CommitHash) {
 	}
 }
 
+const startingContainers: Map<CommitHash, Promise<ContainerInfo> > = new Map();
 export async function startContainer(commitHash: CommitHash) {
 	l.log({ commitHash }, `Starting up container for ${commitHash}`);
 	const image = getImageName(commitHash);
+
+	// do we have an existing container?
+	const existingContainer = getRunningContainerForHash( commitHash );
+	if ( existingContainer ) {
+		l.log( { commitHash, containerId: existingContainer.Id }, `Found a running container for ${commitHash}`);
+		return Promise.resolve( existingContainer );
```

Comment:
I'm not sure we really need to return the container info, but it has a benefit later